torch/_decomp/decompositions.py

```diff
@@ -2151,14 +2168,7 @@ def upsample_nearest2d(
     )
     result = input[:, :, h_indices, w_indices]
 
-    # convert output to correct memory format, if necessary
-    memory_format = utils.suggest_memory_format(input)
-
-    # following "heuristic: only use channels_last path when it's faster than the contiguous path"
-    _, n_channels, _, _ = input.shape
-    if input.device.type == "cuda" and n_channels < 4:
-        memory_format = torch.contiguous_format
```

Comment:
Removed this heuristics due to failing test_fake_tensor.py tests and also https://github.com/pytorch/pytorch/pull/91260/files#r1183144329